consensus/core/src/storage/mem_store.rs

```diff
@@ -208,6 +260,11 @@ impl Store for MemStore {
         Ok(commits)
     }
 
+    fn scan_misbehavior_reports(&self) -> ConsensusResult<HashSet<MisbehaviorReport>> {
+        let inner = self.inner.read();
+        Ok(inner.misbehavior_reports.clone())
```

Comment:
During recovery, not everything is scanned, in general. I am not sure if reading all `MisbehaviorReport` from storage is doable. I did something similar, initially, for the unprovable metrics, and got reviews saying that this should not be done.

Reply:
Same comment in rocksdb_store.rs